src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
+        return this.icalTimeToLocalMoment(this._dtstart);
     }
 
+    // If set on an event with RRULE, updates entire repeated event
+    // to now *start* from this new date
+    // If set on an exception, only changes that exception
     set dtstart(value: moment.Moment) {
         const allDay = this.allDay;
-        this.event.startDate = ICAL.Time.fromJSDate(value.toDate());
+        this._dtstart = this.momentToIcalTime(value, this.event.startDate ? this.event.startDate.zone : null);
+        // FIXME: Do this only if its the first one?
+        this.event.startDate = this._dtstart;
+        // FIXME: regenerate all following events!?
+        // Do we just let the save-to-dav / reloadEvents cycle do this for us?
         this.allDay = allDay;
     }
 
     get dtend(): moment.Moment {
-        return this.event.endDate
-            ? this.icalTimeToLocalMoment(this.event.endDate)
+        return this._dtend
+            ? this.icalTimeToLocalMoment(this._dtend)
             : undefined;
     }
 
+    // If set on an event with RRULE, updates entire repeated event
+    // ?? Does this shorten it to one day ? (does it include the date)
+    // If set on an exception, only changes that exception
+    // FIXME: Does this change an ICAL.Time with no date (isDate=true) to
+    // one with a time? (as a side effect that is)
+    // See also set dtstart comments!
     set dtend(value: moment.Moment) {
         const allDay = this.allDay;
-        this.event.endDate = value ? ICAL.Time.fromJSDate(value.toDate()) : undefined;
+        if (value) {
+            this._dtend = this.momentToIcalTime(value, this.event.endDate ? this.event.endDate.zone : undefined);
+            // FIXME: (see above)
+            this.event.endDate = this._dtend;
+        }
         this.allDay = allDay;
     }
 
     // angular-calendar compatibility
 
     get start(): Date {
-        return this.dtstart.toDate();
+        return this._dtstart.toJSDate();
     }
 
     get end(): Date {
-        if (!this.dtend) {
+        if (!this._dtend) {
             return undefined;
         }
 
-        const shownEnd = moment(this.dtend);
+        const shownEnd = this._dtend.clone();
+        // ICAL event DTEND is exclusive, angular-calendar is inclusive
         if (this.allDay) {
-            shownEnd.subtract(1, 'days');
+            shownEnd.addDuration(new ICAL.Duration({'isNegative': true, 'days': 1}));
         } else {
-            shownEnd.subtract(1, 'seconds');
+            shownEnd.addDuration(new ICAL.Duration({'isNegative': true, 'seconds': 1}));
         }
-        return shownEnd.toDate();
+        return shownEnd.toJSDate();
     }
 
     get allDay(): boolean {
         // isDate in ICAL.Event means "has no time"
-        return this.event.startDate.isDate;
+        return this._dtstart.isDate;
+    }
+
+    get recurs(): boolean {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        return recur ? true : false;
```

Comment:
Seems like we could just skip this variable entirely :)

Reply:
I am using it in quite a few places..

Reply:
Ah, I was confused and assumed `getFirstProperlyVaule` returns a boolean which we could just return. We could swap this whole body for something like `return !!this.event.component.getFirstPropertyValue('rrule');`, though that doesn't really make it any more readable :)